lib/AST/DistributedDecl.cpp

```diff
@@ -524,23 +524,30 @@ bool AbstractFunctionDecl::isDistributedActorSystemRemoteCall(bool isVoidReturn)
     }
   }
 
-  // === Check generic parameters in detail
   // --- Check: Act: DistributedActor,
   //            Act.ID == Self.ActorID
-  GenericTypeParamDecl *ActParam = genericParams->getParams()[0];
-  auto ActConformance = module->lookupConformance(
-      mapTypeIntoContext(ActParam->getDeclaredInterfaceType()),
-      C.getProtocol(KnownProtocolKind::DistributedActor));
-  if (ActConformance.isInvalid()) {
-    return false;
-  }
+  auto *ActParam = std::find_if_not(
+      genericParams->begin(), genericParams->end(),
+      [this, &C, &module](GenericTypeParamDecl *param) {
+        return module
+            ->lookupConformance(
+                mapTypeIntoContext(param->getDeclaredInterfaceType()),
+                C.getProtocol(KnownProtocolKind::DistributedActor))
+            .isInvalid();
```

Comment:
Instead of doing it this way maybe getting it form the parameter and verifying that the conformance is there is a easier approach because even though we don't know what order they are declared in generic signature, they should be used in a particular way?...

Reply:
Hmm yeah that might be more reliable, find it from the parameter position and find the generic... I'll give that a go! Thanks!

Reply:
Nice, I managed to clean things up a bit, thank you @xedin !